microsoft/surface/common/ipts/default.nix

```diff
@@ -41,8 +41,12 @@ in
         path = with pkgs; [ iptsd ];
         script = "iptsd $(iptsd-find-hidraw)";
         wantedBy = [ "multi-user.target" ];
+        restartTriggers = [ (if (cfg.config != null) then iptsConfFile else "") ];
```

Comment:
I'm not sure if this is the 'proper' way, the values in `restartTriggers` need to be strings from the errors I got, so now we return an empty string or the path to the config file.

Reply:
This is better:

```suggestion
        restartTriggers = lib.optional (cfg.config != null) iptsConfFile;
```

Reply:
Applied the suggestion, thanks for showing me the better way!